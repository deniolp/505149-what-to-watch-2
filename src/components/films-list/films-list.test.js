import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

import FilmsList from './films-list';
import filmsListMock from '../../mocks/films';

describe(`FilmsList`, () => {
  const filmsMock = filmsListMock.slice(0, 2);

  it(`renders correctly`, () => {
    const tree = renderer.create(
        <BrowserRouter><FilmsList
          films={filmsMock}
          filmsCounter={2}
        /></BrowserRouter>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
