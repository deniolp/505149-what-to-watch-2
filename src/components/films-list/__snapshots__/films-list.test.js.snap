// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsList renders correctly 1`] = `
Array [
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <a
      href="/film/1"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Aviator"
          height="175"
          src="img/aviator.jpg"
          width="280"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/film/1"
        onClick={[Function]}
      >
        Aviator
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <a
      href="/film/2"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Shutter Island"
          height="175"
          src="img/shutter-island.jpg"
          width="280"
        />
      </div>
    </a>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/film/2"
        onClick={[Function]}
      >
        Shutter Island
      </a>
    </h3>
  </article>,
]
`;
